scripts/update-component-sha.sh

```diff
@@ -1,10 +1,67 @@
 #!/bin/sh
 set -e
-if [ $# -ne 2 ] ; then
-    echo "Need <OLD> and <NEW> as arguments" >&2
+
+##
+#
+# script to replace hashes in config files
+# see usage() for usage and functionality
+#
+
+function usage() {
+    cat >&2 <<EOF
+$0 --<mode> <how-to-find> <new-hash>
+
+Available modes: --hash and --image
+
+Replace by hash:
+	$0 --hash <OLD> <NEW>
+	Example: $0 8675309abcdefg abcdef567899
+    	   Will replace all instances of 8675309abcdefg with abcdef567899
+
+Replace by image: $0 --image <IMAGE> <NEW>
+	$0 --image <IMAGE> <NEW>
+	Example: $0 linuxkit/foo abcdef567899
+	   Will tag all instances of linuxkit/foo with abcdef567899
+
+By default, for convenience, if no mode is given (--image or --hash), the furst method (--hash) is assumed. Thus the following are equivalent:
```

Comment:
It has a hidden meaning. The first method actually is the prince or ruler method!